gamestonk_terminal/portfolio/portfolio_controller.py

```diff
@@ -143,6 +184,247 @@ def call_po(self, _):
+            dest="name",
+            required="-h" not in other_args,
+            help="Name of file to be saved",
+        )
+        if other_args:
+            if "-n" not in other_args and "-h" not in other_args:
+                other_args.insert(0, "-n")
+
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        self.portfolio = portfolio_model.load_df(ns_parser.name)
+        print("")
+
+    @try_except
+    def call_save(self, other_args: List[str]):
+        """Process save command"""
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            prog="save",
+            description="Save your portfolio",
+        )
+        parser.add_argument(
+            "-n",
+            "--name",
+            type=str,
+            dest="name",
+            required="-h" not in other_args,
+            help="Name of file to be saved",
+        )
+        if other_args and "-n" not in other_args and "-h" not in other_args:
+            other_args.insert(0, "-n")
+
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        if (
+            ".csv" not in ns_parser.name
+            and ".xlsx" not in ns_parser.name
+            and ".json" not in ns_parser.name
+        ):
+            print(
+                "Please submit as 'filename.filetype' with filetype being csv, xlsx, or json\n"
+            )
+
+        portfolio_model.save_df(self.portfolio, ns_parser.name)
+
+    def call_show(self, _):
+        """Process show command"""
+        portfolio_view.show_df(self.portfolio, False)
+
+    def call_add(self, other_args: List[str]):
+        """Process add command"""
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            prog="add",
+            description="Adds an item to your portfolio",
+        )
+        parser.add_argument(
+            "-n",
+            "--name",
+            type=str,
+            dest="name",
+            required="-h" not in other_args,
+            help="Name of item to be added (for example a ticker for a stock)",
+        )
+        parser.add_argument(
+            "-t",
+            "--type",
+            dest="type",
+            type=str,
+            choices=["stock", "bond", "option", "crypto", "cash"],
+            default="stock",
+            help="Type of asset to add",
+        )
+        parser.add_argument(
+            "-q",
+            "--quantity",
+            dest="quantity",
+            type=float,
+            default=1,
+            help="Amounts of the asset owned",
+        )
+        parser.add_argument(
+            "-d",
+            "--date",
+            dest="date",
+            type=str,
+            default=datetime.now().strftime("%Y/%m/%d"),
```

Comment:
This is not the type of data you want. The code crashes if I add something like this:
```
(✨) (portfolio)> ar
'<' not supported between instances of 'str' and 'Timestamp'
```

The reason may be that everywhere in the terminal we use YYYY-mm-dd. So we should keep with that I'd say.